datadotworld/client/api.py

```diff
@@ -1583,15 +1544,33 @@ def __build_dataset_obj(dataset_constructor, file_constructor, args):
         return dataset
 
     @staticmethod
-    def __build_project_obj(project_constructor, file_constructor, args):
+    def __build_dataset_obj_no_files(dataset_constructor, args):
```

Comment:
you can probably DRY this up since `__build_dataset_obj_no_files` is a superset of `__build_dataset_obj` now. I'm thinking something like

```python
@staticmethod
def __build_dataset_obj(dataset_constructor, file_constructor, args):
    dataset = __build_dataset_obj_no_files(dataset_constructor, args)
    # ...the files = ([file_constructor(...)]) bits
    dataset.files = files
```